src/Native/System.Native/pal_mount.cpp

```diff
@@ -101,6 +101,13 @@ extern "C" int32_t SystemNative_GetSpaceInfoForMountPoint(const char* name, Moun
     return result;
 }
 
+typedef decltype(statfs::f_type) typeof_statfs_f_type;
+
+int64_t GetFType(typeof_statfs_f_type statfs_f_type)
```

Comment:
Thanks. Will change the name.

> right now I wouldn't expect the function to cast

I think the compiler optimizes this code and squashes the `static_cast`s when the conversion is redundant and ultimately it has no overhead.